### ISSUE TITLE
[FEATURE] Google Tag Manager Integration

### ISSUE THREAD

**Issue by jobenjada** [formbricks/formbricks#1983]
### Is your feature request related to a problem? Please describe.

tbd

### Describe the solution you'd like

tbd

**Comment by jonas-hoebenreich** [comment]
Uuh, very interesting topic. Is this about the implementation of GTM in the backend or in the frontend? If it's about an implementation in the frontend library, I would have a few thoughts on this and would also be willing to create a PR :)

**Comment by mattinannt** [comment]
Currently not a priority since not requested a lot for experience management use cases and the privacy topic makes it hard to implement properly on Formbricks Cloud.

**Comment by alcoolio** [comment]
i think anyone using this in a corporate environment would need GTag implementatrion at some point.

basically we are just asking for java script implementation in steps and thankyou-page :)

**Comment by jobenjada** [comment]
Hey @alcoolio thanks for following up :) What do you mean with JS implementation "in steps"? You mean questions?

The Privacy Issue remains a concern due to our positioning as the most privacy-friendly solution but I have resurfaced this in the product requests so we will talk about it in our next Product Meeting.

**Comment by alcoolio** [comment]
Hey @jobenjada 
thanks for your quick reply! 

Yes I mean a Gtag covering all questions/pages/steps and then a Conversion Pixel for the thank you page.

I have chosen Formbricks also for the privacy and for the selfhosting option and I think people do love it for that. But still for some of my clients I would still need it to run ads properly - and there you need to send back those signals.

As this is an optional way to use FB, I am not sure if it would endanger the privacy issue. 

We could still enforce cookie consent usage or maybe there is a way to prohibit Google from using the forms user data for tracking?

**Comment by jobenjada** [comment]
Yeah the cookie consent thingy was the issue mostly and we'd need to handle it differently between on prem and Cloud because in one case we as a company are the data processor and in another case we are not (but still need to provide self-hosters with an option to use GTag in a compliant way).

Can you share a bit more about your use cases and customers? Always curious to learn more! And if they match our Ideal Customer Profile we are much more likely to ship this quickly. Feel free to drop me an email at johannes@formbricks.com
